modcc/functioninliner.cpp

```diff
@@ -76,9 +59,78 @@ expression_ptr inline_function_call(Expression* e)
                                      e->location());
         }
 
-        return new_e;
+        if (e->is_assignment()) {
+            e->is_assignment()->replace_lhs(lhs->clone());
+        }
+    };
+
+    if (exp->is_if()) {
+        if (!exp->is_if()->false_branch()) {
+            throw compiler_exception(
+                    "can only inline if statements with associated else", exp->location()
+            );
+        }
+        // Modify the condition
+        fix_expression(exp);
+
+        auto& true_branch = exp->is_if()->true_branch()->is_block()->statements();
+        auto& false_branch = exp->is_if()->false_branch()->is_block()->statements();
+
+        if (true_branch.size() != 1 || false_branch.size() != 1) {
+            throw compiler_exception(
+                    "can only inline functions with one statement", exp->location()
+            );
+        }
+
+        if (true_branch.front()->is_if()) {
+            replace_and_inline(true_branch.front(), lhs, fargs, cargs, scope);
+        } else if (true_branch.front()->is_assignment()) {
+            // Modify the true assignment
+            fix_expression(true_branch.front());
+        } else {
+            throw compiler_exception(
+                    "can only inline assignment expressions and if expressions containing single assignment expressions", exp->location()
+            );
+        }
+
+        if (false_branch.front()->is_if()) {
+            replace_and_inline(false_branch.front(), lhs, fargs, cargs, scope);
```

Comment:
Does this recursively handle the case of multiple `else if` statements?

Are both of these cases now handled?

```
if (x) {
}
elif (y) {
}
elif (z) {
}
else {
}
```

```
if (x) {
  if (y) {
  }
  else {
  }
}
else {
}
```

Reply:
The second case is handled. The first is not.